src/ngraph/pass/pass_config.hpp

```diff
@@ -23,21 +23,28 @@ namespace ngraph
     namespace pass
     {
         class PassConfig;
+        enum class CompilationMode
+        {
+            DEX,    // Fast compilation using precompiled kernels
+            CODEGEN // Slower compilation for potentially faster code
+        };
     }
 }
 
 class ngraph::pass::PassConfig
 {
 public:
-    PassConfig();
+    PassConfig(CompilationMode mode = CompilationMode::DEX);
     const std::map<std::string, bool>& get_enables() { return m_pass_enables; }
     void set_pass_enable(std::string name, bool enable);
     bool get_pass_enable(std::string name);
     const std::map<std::string, bool>& get_pass_attributes() { return m_pass_attributes; }
     void set_pass_attribute(std::string name, bool enable);
     bool get_pass_attribute(std::string name);
-
+    CompilationMode get_compilation_mode() const { return m_compilation_mode; }
+    void set_compilation_mode(CompilationMode mode) { m_compilation_mode = mode; }
```

Comment:
I wonder if we should remove the setter so that we only allow setting the compilation mode through the constructor. This would prevent clients to change it randomly. We might have initialization based on the compilation mode in the future.

Reply:
i'm ok either way. i can remove it for now and we can add later if needed.